tasks/deploy/github.py

```diff
@@ -0,0 +1,157 @@
+        """
+        Checks if a repo can be found. This is useful for testing access permissions to repos.
+        """
+        result = self.repo(repo)
+
+        # name is arbitrary, just need to check if something is in the result
+        if "name" in result:
+            return
+
+        print("Cannot find Github repo {}".format(repo))
+        print("If you cannot see it in the Github UI, you likely need permission.")
+        raise Exit(code=1)
+
+    def repo(self, repo_name):
+        """
+        Gets the repo info.
+        """
+
+        path = "/repos/{}".format(repo_name)
+        return self.make_request(path, json=True)
+
+    def trigger_workflow(self, repo_name, workflow_name, ref, inputs=None):
+        """
+        Create a pipeline targeting a given reference of a project.
+        ref must be a branch or a tag.
+        """
+        if inputs is None:
+            inputs = {}
+
+        path = "/repos/{}/actions/workflows/{}/dispatches".format(repo_name, workflow_name)
+        data = json.dumps({"ref": ref, "inputs": inputs})
+        return self.make_request(path, data=data)
+
+    def workflow_run(self, repo_name, run_id):
+        """
+        Gets info on a specific workflow.
+        """
+        path = "/repos/{}/actions/runs/{}".format(repo_name, run_id)
+        return self.make_request(path, json=True)
+
+    def download_artifact(self, repo_name, artifact_id, destination_dir):
+        """
+        Downloads the artifact identified by artifact_id to destination_dir.
+        """
+        path = "/repos/{}/actions/artifacts/{}/zip".format(repo_name, artifact_id)
+        content = self.make_request(path, raw_content=True)
+
+        zip_target_path = os.path.join(destination_dir, "{}.zip".format(artifact_id))
+        with open(zip_target_path, "wb") as f:
+            f.write(content)
+        return zip_target_path
+
+    def workflow_run_artifacts(self, repo_name, run_id):
+        """
+        Gets list of artifacts for a workflow run.
+        """
+        path = "/repos/{}/actions/runs/{}/artifacts".format(repo_name, run_id)
+        return self.make_request(path, json=True)
+
+    def latest_workflow_run_for_ref(self, repo_name, workflow_name, ref):
+        """
+        Gets latest workflow run for a given reference
+        """
+        runs = self.workflow_runs(repo_name, workflow_name)
+        ref_runs = [run for run in runs["workflow_runs"] if run["head_branch"] == ref]
+
+        if len(ref_runs) == 0:
+            return None
+
+        return sorted(ref_runs, key=lambda run: run['created_at'], reverse=True)[0]
+
+    def workflow_runs(self, repo_name, workflow_name):
+        """
+        Gets all workflow runs for a workflow.
+        """
+        path = "/repos/{}/actions/workflows/{}/runs".format(repo_name, workflow_name)
+        return self.make_request(path, json=True)
+
+    def make_request(self, path, headers=None, data=None, json=False, raw_content=False):
```

Comment:
Would you mind documenting what the parameters? This function is probably the most important and it would make it easier to read the code if it was documented.

Also, a nit (feel free to ignore): I think it would be better to have an argument like `output_format` that can have either `json`, `raw` or `text` (the default) instead of having separate arguments (since right now there are combinations that don't make much sense like `json=True, raw_content=True`)